### PR TITLE
Cleanup listener wrapping in TransportSearchAction a little

### PR DESCRIPTION
**Pull request by original-brownbear**
Lets avoid parsing settings during search request processing, remove unused field from the listener and using the delegating listener class.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java [hunk 1]
```diff
@@ -193,9 +193,10 @@ public TransportSearchAction(
         this.indexNameExpressionResolver = indexNameExpressionResolver;
         this.namedWriteableRegistry = namedWriteableRegistry;
         this.executorSelector = executorSelector;
-        this.settings = clusterService.getSettings();
+        var settings = clusterService.getSettings();
         this.defaultPreFilterShardSize = DEFAULT_PRE_FILTER_SHARD_SIZE.get(settings);
         this.ccsCheckCompatibility = SearchService.CCS_VERSION_CHECK_SETTING.get(settings);
+        this.collectTelemetry = SearchService.CCS_COLLECT_TELEMETRY.get(settings);
```
Comment (smalyshev) [inline server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java:199]: Wouldn't this mean if the setting changes in runtime, it won't be picked up here?

Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java:199]: It wasn't picked up before either :) If this was supposed to be a cluster setting that can be runtime adjusted then it needs to be implemented as one. `Settings` themselves are 100% immutable.

Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java:199]: That's the same thing, `settings` is a final field in the cluster service also. If we want dynamic updates here then we need to use the machinery around `org.elasticsearch.common.settings.ClusterSettings`.

Comment (smalyshev) [reply server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java:199]: I think I do want it to be dynamic, doesn't have to be in this pull though.

**Review by smalyshev — APPROVED** [review]

**Pull request merged**